Python/ceval.c

```diff
@@ -4645,14 +4627,16 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
             PyObject *function = PEEK(oparg + 1);
             if (Py_TYPE(function) == &PyFunction_Type) {
                 PyCodeObject *code = (PyCodeObject*)PyFunction_GET_CODE(function);
-                STACK_SHRINK(oparg + 1);
+                PyObject *locals = code->co_flags & CO_OPTIMIZED ? NULL : PyFunction_GET_GLOBALS(function);
                 if ((code->co_flags & (CO_GENERATOR | CO_COROUTINE | CO_ASYNC_GENERATOR)) == 0) {
-                    InterpreterFrame *new_frame = _PyEval_FrameFromPyFunctionAndArgs(tstate, stack_pointer+1, oparg, function);
+                    InterpreterFrame *new_frame = _PyEvalFramePushAndInit(
```

Comment:
Aren't we loosing here a ton of asserts that were in `_PyEval_FrameFromPyFunctionAndArgs`?

Reply:
They weren't asserting anything meaningful or that isn't re-asserted elsewhere.
All the `nargsf` stuff was just asserting truisms, and there's not much point in
`assert(!(code->co_flags & (CO_GENERATOR | CO_COROUTINE | CO_ASYNC_GENERATOR)));`
immediately after
`if ((code->co_flags & (CO_GENERATOR | CO_COROUTINE | CO_ASYNC_GENERATOR)) == 0) {`